PSKoans/Koans/Constructs and Patterns/AboutBitwiseOperations.Koans.ps1

```diff
@@ -0,0 +1,545 @@
+                    ------------------+-----------+----------+----------+----------+
+                    Bitwise operation | -shl 24   | -shl 16  | -shl 8   |          |
+                    ------------------+----------=+----------+----------+----------+
+                    Int32 Value       | 369098752 | 3670016  | 49152    | 15       |
+
+                The final Int32 value, 372817935, is obtained by adding each of the individual pieces together.
+            #>
+
+            # Hint: set one bit for each value
+            $BinaryValues = '________', '________', '________', '________'
+
+            $Bytes = $BinaryValues | ConvertFrom-Binary
+
+            [Int]$Bytes[0] -shl 24 | Should -Be 1073741824
+            [Int]$Bytes[1] -shl 16 | Should -Be 2097152
+            [Int]$Bytes[2] -shl 8 | Should -Be 4096
+            $Bytes[3] | Should -Be 8
+
+            $Value = @(
+                [Int]$Bytes[0] -shl 24
+                [Int]$Bytes[1] -shl 16
+                [Int]$Bytes[2] -shl 8
+                $Bytes[3]
+            ) | Measure-Object -Sum
+
+            $Value.Sum | Should -Be 1652126821
+            -join $BinaryValues | ConvertFrom-Binary -To Int32 | Should -Be $Value.Sum
+        }
+    }
+
+
+    Context 'About the signing bit' {
+        <#
+            A byte is an example of an unsigned integer. The values can range from 0 to 255, the value cannot
+            be a negative number.
+
+            Signed integers reserve a single bit to denote whether the value is positive or negative.
+            When the value of the signing bit is 0, the value is positive. When the value is 1, the value is
+            negative.
+
+            The signing bit takes the place of the most significant bit.
+
+                    Bit position      Value
+                    ------------      -----
+                               1      Signing bit
+                               2      64
+                               3      32
+                               4      16
+                               5      8
+                               6      4
+                               7      2
+                               8      1
+
+            This value is known as SByte, or signed byte. The signed byte allows numbers in the range -128 to 127.
+
+            In .NET, SByte, Int16, Int32, and Int64 are all signed integers. Byte, UInt16, UInt32, and UInt64 are all
+            unsigned integers.
+        #>
+
+        It 'allows positive values to be expressed in the same way as an unsigned integer' {
+            # Positive values do not use the signing bit.
+
+            '0_______' | ConvertFrom-Binary -To SByte | Should -Be (1 + 4 + 8)
+        }
+
+        It 'uses the most significant bit to describe a negative number' {
+            # When the signing bit alone is set, the value is the smallest value, -128.
+
+            '1000000' | ConvertFrom-Binary -To SByte | Should -Be -128
```

Comment:
```suggestion
            '1000000' | ConvertFrom-Binary -To SByte | Should -Be (-128)
```
This will get parsed as a string argument otherwise.... and I'm not sure if that's a problem or not. Best to be safe? 🙂

Reply:
Yeah, I thought it looked a bit off, and had to double check what it was doing. Thanks!